sciLifeLab_utils/run_QC_analysis.py

```diff
@@ -47,7 +47,17 @@ def main(args):
         flowcells_dirs  = [os.path.join(sample_data_dir,flowcell) \
                 for flowcell in os.listdir(sample_data_dir) \
                 if os.path.isdir(os.path.join(sample_data_dir,flowcell))]
-
+        
+        # to adapt the diretory structure in IRMA where it have one folder for lib prep.
+        # So shcek if the collected FC directories are really FC directory or a LIB prep dir
+        lib_prep_dirs = [fc for fc in flowcells_dirs if re.match(r'^[A-Z]$', os.path.basename(fc))]
+        for prep_dir in lib_prep_dirs:
+            flowcells_dirs.extend([os.path.join(prep_dir,flowcell) \
+                for flowcell in os.listdir(prep_dir) \
+                if os.path.isdir(os.path.join(prep_dir,flowcell))])
+            if prep_dir in flowcells_dirs:
+                flowcells_dirs.remove(prep_dir)
```

Comment:
the two ways of build flowcells_dirs are mutually exclusive and they are based on different assumption on the file system structure. So it would be better to not populate flowcells_dirs the first time but to perform a sanity check on which format we have